modules/Auth/Services/AuthService.php

```
@@ -162,28 +161,28 @@ private function sendOtpByStep($step, $identifier)
 
     private function checkOtpByStep($step, $identifier, $otp)
     {
-        if (!property_exists($step, 'login_option')) {
+        if (empty($step)) {
             return true;
         }
-
         if ($step->login_option == "otp") {
             return (new Otp)->validate($identifier, $otp)->status;
         }
-
         return true;
     }
 
-    // private function checkOtpByStep($step, $identifier, $otp)
-    // {
-    //     if ($step->login_option == "otp" && (new Otp)->validate($identifier, $otp)->status == false) {
-    //         throw new \Exception(__("validation.invalid-otp"), 401);
-    //     }
-    //     return true;
-    // }
+
     private function checkPasswordByStep($step, $identifier, $password)
     {
         $user = $this->userCRUDService->getUserByIdentifier($identifier);
 
+        if ($step === null || !is_object($step) || get_class($step) === 'stdClass') {
+            return false;
+        }
+
+        if (!property_exists($step, 'login_option')) {
```

The `property_exists` check should be moved before the password verification logic to fail fast if the step is invalid.